### PR TITLE
Kill node on failed policy list Load and other fixes

### PR DESCRIPTION
**Pull request by nkostoulas**

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -1734,15 +1734,27 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
...
-        LogPrintf("freezelist...\n");
-        if (fRequireFreezelistCheck) LoadFreezeList(pcoinsTip);
-        LogPrintf("burnlist...\n");
-        if (fEnableBurnlistCheck) LoadBurnList(pcoinsTip);
-        LogPrintf("whitelist.\n");
-        if (fRequireWhitelistCheck || fScanWhitelist) {
-	       addressWhitelist->Load(pcoinsTip);
-	    }
-        if (fRequestList) requestList.Load(pcoinsTip, chainActive.Height());
+        if (fRequireFreezelistCheck) {
```
Comment (lawlawlaw) [inline src/init.cpp]: Wouldn't it be better to have these functions throw exceptions with more debug info?

Comment (nkostoulas) [reply src/init.cpp]: I guess so but currently the only error that can happen is when reading from the utxo db fails and this is already reported by `return error("%s: unable to read value", __func__);`.

Comment (nkostoulas) [reply src/init.cpp]: That's the same for all the "Load..." methods.

Comment (lawlawlaw) [reply src/init.cpp]: There's one more for whitelist which is if recoverwhitelistkeys is enabled and that fails

Comment (lawlawlaw) [reply src/init.cpp]: But otherwise that's fine

Comment (nkostoulas) [reply src/init.cpp]: Ah right I'll add that too

**Pull request merged**